ksqldb-execution/src/main/java/io/confluent/ksql/execution/interpreter/CastInterpreter.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.interpreter;
+
+import io.confluent.ksql.execution.codegen.helpers.CastEvaluator;
+import io.confluent.ksql.execution.interpreter.terms.CastTerm;
+import io.confluent.ksql.execution.interpreter.terms.CastTerm.CastFunction;
+import io.confluent.ksql.execution.interpreter.terms.Term;
+import io.confluent.ksql.schema.ksql.SqlBooleans;
+import io.confluent.ksql.schema.ksql.SqlDoubles;
+import io.confluent.ksql.schema.ksql.SqlTimestamps;
+import io.confluent.ksql.schema.ksql.types.SqlArray;
+import io.confluent.ksql.schema.ksql.types.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlDecimal;
+import io.confluent.ksql.schema.ksql.types.SqlMap;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.util.DecimalUtil;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public final class CastInterpreter {
+  private CastInterpreter() { }
+
+  public static CastTerm cast(
+      final Term term,
+      final SqlType from,
+      final SqlType to,
+      final KsqlConfig config
+  ) {
+    return new CastTerm(term, to, castFunction(from, to, config));
+  }
+
+  private static CastFunction castFunction(
+      final SqlType from,
+      final SqlType to,
+      final KsqlConfig config
+  ) {
+    final SqlBaseType toBaseType = to.baseType();
+    if (toBaseType == SqlBaseType.INTEGER) {
+      return castToInteger(from);
+    } else if (toBaseType == SqlBaseType.BIGINT) {
+      return castToLong(from);
+    } else if (toBaseType == SqlBaseType.DOUBLE) {
+      return castToDouble(from);
+    } else if (toBaseType == SqlBaseType.DECIMAL) {
+      return castToBigDecimal(from, to);
+    } else if (toBaseType == SqlBaseType.STRING) {
+      return castToString(from, config);
+    } else if (toBaseType == SqlBaseType.BOOLEAN) {
+      return castToBoolean(from);
+    } else if (toBaseType == SqlBaseType.ARRAY) {
+      return castToArrayFunction(from, to, config);
```

Comment:
nit: why add `Function` suffix for these two?

Reply:
Good question.  That's a holdover from when it actually did it rather than compiled it.  Changed to all have suffix `Function`.